src/orion/algo/tpe.py

```diff
@@ -586,13 +575,20 @@ def _build_mixture(self):
         for mu, sigma in zip(self.mus, self.sigmas):
             self.pdfs.append(norm(mu, sigma))
 
-    def sample(self, num=1):
+    def sample(self, num=1, attempts=10):
         """Sample required number of points"""
         point = []
         for _ in range(num):
             pdf = numpy.argmax(self.tpe.rng.multinomial(1, self.weights))
-            new_points = self.pdfs[pdf].rvs(size=5, random_state=self.tpe.rng)
-            for pt in new_points:
+            new_points = list(
+                self.pdfs[pdf].rvs(size=attempts, random_state=self.tpe.rng)
+            )
+            while True:
+                if not new_points:
+                    raise RuntimeError(
+                        f"Failed to sample in interval ({self.low}, {self.high})"
```

Comment:
It is still possible if user just re-run the same experiment after get this `RuntimeError`, right?

Reply:
Yes, but the user will likely hit this error again. Basically if the interval gets to narrow it means the algorithm has pretty much converged. It may not be true for all dimensions however.